src/container-manager.ts

```diff
@@ -0,0 +1,50 @@
+import { spawn, spawnSync } from 'child_process'
+import { writeFileSync } from 'fs';
+import { mkdtemp, rm } from 'fs/promises';
+import path = require('path');
+
+export class ContainerManager {
+    constructor(public command: string = "docker") { }
+
+    async push(imageName: string, auth: string) {
```

Comment:
This abstraction feels little leaky to me in that it expects `imageName` to be an image that's been prefixed with a registry path. Maybe that's fine, but then it seems like it's really tailored for the one spot that it's used, in which case, I wonder if factoring it out into this `ContainerManager` class does much for us. Happy to leave it like this if you feel strongly about it, but I kinda think it's simpler with push and tag as just functions defined where they're used. Just my two cents though!

Reply:
Ah yes, good point about `docker push` already expecting the registry prefix